File: easy school/fees/fees rates.vb
Public Class fees_rates
    Dim data As New database
    Dim level, feety As New List(Of String)
    Dim lev, cla, ikk As String
    Private Sub fees_rates_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim red As New DataTable
        red = data.executeSQL("SELECT `code`,  `description` FROM `class` ORDER BY `level`")  'subjects
        For Each drow In red.Rows
            ComboBox1.Items.Add(drow.Item(1).ToString.ToUpper)
            level.Add(drow.Item(0).ToString.ToUpper)
        Next
        Dim red1 As New DataTable
        red1 = data.executeSQL("SELECT `ID`, `Type_Name` FROM `type`")  'subjects
        For Each drow In red1.Rows
            ComboBox2.Items.Add(drow.Item(1).ToString.ToUpper)
            feety.Add(drow.Item(0).ToString.ToUpper)
        Next
       
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        If CheckBox1.Checked = True Then
            Dim j As Integer
            For Each b As String In level
                data.add("INSERT INTO `rates`( `L_id`, `T_id`, `amount`) VALUES ('" & b & "','" & cla & "'," & TextBox1.Text & ")")
                j = j + 1
            Next
            Dim ggf As DataTable
            ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates")
            DataGridView1.DataSource = ggf
            Dim i, sum As Integer
            Do Until i = DataGridView1.Rows.Count
                sum = sum + DataGridView1.Rows(i).Cells(3).Value
                i = i + 1
            Loop
            Label3.Text = "Total = Ksh " & sum
            Exit Sub
        End If

        'dsfgg
        If ComboBox1.Text = "" Or ComboBox1.SelectedIndex < 0 Then
            MsgBox("data required", MsgBoxStyle.Information, "error")
            ComboBox1.Focus()
            Exit Sub
        Else
            If ComboBox2.Text = "" Or ComboBox2.SelectedIndex < 0 Then
                MsgBox("data required", MsgBoxStyle.Information, "error")
                ComboBox2.Focus()
                Exit Sub
            Else
                If TextBox1.Text = "" Then
                    MsgBox("data required", MsgBoxStyle.Information, "error")
                    TextBox1.Focus()
                    Exit Sub
                Else
                    data.add("INSERT INTO `rates`( `L_id`, `T_id`, `amount`) VALUES ('" & lev & "','" & cla & "'," & TextBox1.Text & ")")
                    Dim ggf As DataTable
                    ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates WHERE rates.L_id =" & lev)
                    DataGridView1.DataSource = ggf
                    Dim i, sum As Integer
                    Do Until i = DataGridView1.Rows.Count
                        sum = sum + DataGridView1.Rows(i).Cells(3).Value
                        i = i + 1
                    Loop
                    Label3.Text = "Total = Ksh " & sum
                End If
            End If
        End If
    End Sub

    Private Sub ComboBox1_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ComboBox1.SelectedIndexChanged
        lev = level(ComboBox1.SelectedIndex)
        Dim ggf As DataTable
        ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates WHERE rates.L_id =" & lev)
        DataGridView1.DataSource = ggf
        Dim i, sum As Integer
        Do Until i = DataGridView1.Rows.Count
            sum = sum + DataGridView1.Rows(i).Cells(3).Value
            i = i + 1
        Loop
        Label3.Text = "Total = Ksh " & sum
    End Sub

    Private Sub ComboBox2_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ComboBox2.SelectedIndexChanged
        cla = feety(ComboBox2.SelectedIndex)
    End Sub

    Private Sub Button3_Click(sender As Object, e As EventArgs) Handles Button3.Click
        On Error Resume Next
        Dim i As Integer
        Dim h As String
        If Button3.Text = "Save" Then
            data.add("UPDATE `rates` SET `amount`=" & TextBox1.Text & " WHERE `ID`=" & ikk)
            ComboBox1.Text = ""
            ComboBox2.Text = ""
            TextBox1.Text = ""
            ComboBox1.Enabled = True
            ComboBox2.Enabled = True
            Button3.Text = "&Modify F3"
            Button2.Enabled = True
            lev = level(ComboBox1.SelectedIndex)
            Dim ggf As DataTable
            ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates WHERE rates.L_id =" & lev)
            DataGridView1.DataSource = ggf
            Dim j, sum As Integer
            Do Until j = DataGridView1.Rows.Count
                sum = sum + DataGridView1.Rows(i).Cells(3).Value
                j = j + 1
            Loop
            Label3.Text = "Total = Ksh " & sum
            Exit Sub
        End If
        If DataGridView1.RowCount > 0 Then
            i = DataGridView1.CurrentRow.Index
            With DataGridView1
                h = .Item(0, i).Value
                ikk = h
                ComboBox1.Text = .Item(1, i).Value
                ComboBox2.Text = .Item(2, i).Value
                TextBox1.Text = .Item(3, i).Value
                ComboBox1.Enabled = False
                ComboBox2.Enabled = False
                Button3.Text = "Save"
                Button2.Enabled = False
            End With
        End If

    End Sub

    Private Sub Button6_Click(sender As Object, e As EventArgs) Handles Button6.Click
        On Error Resume Next
        lev = level(ComboBox1.SelectedIndex)
        Dim ggf As DataTable
        ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates WHERE rates.L_id =" & lev)
        DataGridView1.DataSource = ggf
        Dim i, sum As Integer
        Do Until i = DataGridView1.Rows.Count
            sum = sum + DataGridView1.Rows(i).Cells(3).Value
            i = i + 1
        Loop
        Label3.Text = "Total = Ksh " & sum
    End Sub

    Private Sub TextBox1_KeyDown(sender As Object, e As KeyEventArgs) Handles TextBox1.KeyDown
       
    End Sub

    Private Sub TextBox1_KeyPress(sender As Object, e As KeyPressEventArgs) Handles TextBox1.KeyPress
        Dim val As New validate
        val.digits(sender, e)
    End Sub

    Private Sub TextBox1_TextChanged(sender As Object, e As EventArgs) Handles TextBox1.TextChanged

    End Sub

    Private Sub Button4_Click(sender As Object, e As EventArgs) Handles Button4.Click
        Dim i As Integer
        Dim h As String
        If DataGridView1.RowCount > 0 Then
            i = DataGridView1.CurrentRow.Index
            With DataGridView1
                h = .Item(0, i).Value
                If MsgBox(" are you sure you want to delete" + vbCrLf + .Item(1, i).Value + " " + .Item(2, i).Value, vbYesNo, "confirm") = vbYes Then
                    data.delete("DELETE FROM `rates` WHERE `ID`=" & h)
                    lev = level(ComboBox1.SelectedIndex)
                    Dim ggf As DataTable
                    ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates WHERE rates.L_id =" & lev)
                    DataGridView1.DataSource = ggf
                End If
            End With
        End If
    End Sub



    Private Sub CheckBox1_CheckedChanged(sender As Object, e As EventArgs) Handles CheckBox1.CheckedChanged
        If CheckBox1.Checked = True Then
            ComboBox1.Enabled = False
        Else
            ComboBox1.Enabled = True
        End If
    End Sub
End Class

File: easy school.ConvertedToC#/fees/fees rates.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Xml.Linq;
using System.Threading.Tasks;
namespace easy_school
{
	public partial class fees_rates
	{
		database data = new database();
		 level;
		List<string> feety = new List<string>();
		string lev;
		string cla;
		string ikk;
		private void fees_rates_Load(object sender, EventArgs e)
		{
			DataTable red = new DataTable();
			red = data.executeSQL("SELECT `code`,  `description` FROM `class` ORDER BY `level`");
			//subjects
			foreach (object drow_loopVariable in red.Rows) {
				drow = drow_loopVariable;
				ComboBox1.Items.Add(drow.Item(1).ToString.ToUpper);
				level.Add(drow.Item(0).ToString.ToUpper);
			}
			DataTable red1 = new DataTable();
			red1 = data.executeSQL("SELECT `ID`, `Type_Name` FROM `type`");
			//subjects
			foreach (object drow_loopVariable in red1.Rows) {
				drow = drow_loopVariable;
				ComboBox2.Items.Add(drow.Item(1).ToString.ToUpper);
				feety.Add(drow.Item(0).ToString.ToUpper);
			}

		}

		private void Button2_Click(object sender, EventArgs e)
		{
			if (CheckBox1.Checked == true) {
				int j = 0;
				foreach (string b in level) {
					data.@add(ref "INSERT INTO `rates`( `L_id`, `T_id`, `amount`) VALUES ('" + b + "','" + cla + "'," + TextBox1.Text + ")");
					j = j + 1;
				}
				DataTable ggf = null;
				ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates");
				DataGridView1.DataSource = ggf;
				int i = 0;
				int sum = 0;
				while (!(i == DataGridView1.Rows.Count)) {
					sum = sum + DataGridView1.Rows[i].Cells[3].Value;
					i = i + 1;
				}
				Label3.Text = "Total = Ksh " + sum;
				return;
			}

			//dsfgg
			if (string.IsNullOrEmpty(ComboBox1.Text) | ComboBox1.SelectedIndex < 0) {
				Interaction.MsgBox("data required", MsgBoxStyle.Information, "error");
				ComboBox1.Focus();
				return;
			} else {
				if (string.IsNullOrEmpty(ComboBox2.Text) | ComboBox2.SelectedIndex < 0) {
					Interaction.MsgBox("data required", MsgBoxStyle.Information, "error");
					ComboBox2.Focus();
					return;
				} else {
					if (string.IsNullOrEmpty(TextBox1.Text)) {
						Interaction.MsgBox("data required", MsgBoxStyle.Information, "error");
						TextBox1.Focus();
						return;
					} else {
						data.@add(ref "INSERT INTO `rates`( `L_id`, `T_id`, `amount`) VALUES ('" + lev + "','" + cla + "'," + TextBox1.Text + ")");
						DataTable ggf = null;
						ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates WHERE rates.L_id =" + lev);
						DataGridView1.DataSource = ggf;
						int i = 0;
						int sum = 0;
						while (!(i == DataGridView1.Rows.Count)) {
							sum = sum + DataGridView1.Rows[i].Cells[3].Value;
							i = i + 1;
						}
						Label3.Text = "Total = Ksh " + sum;
					}
				}
			}
		}

		private void ComboBox1_SelectedIndexChanged(object sender, EventArgs e)
		{
			lev = level(ComboBox1.SelectedIndex);
			DataTable ggf = null;
			ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates WHERE rates.L_id =" + lev);
			DataGridView1.DataSource = ggf;
			int i = 0;
			int sum = 0;
			while (!(i == DataGridView1.Rows.Count)) {
				sum = sum + DataGridView1.Rows[i].Cells[3].Value;
				i = i + 1;
			}
			Label3.Text = "Total = Ksh " + sum;
		}

		private void ComboBox2_SelectedIndexChanged(object sender, EventArgs e)
		{
			cla = feety[ComboBox2.SelectedIndex];
		}

		private void Button3_Click(object sender, EventArgs e)
		{
			 // ERROR: Not supported in C#: OnErrorStatement

			int i = 0;
			string h = null;
			if (Button3.Text == "Save") {
				data.@add(ref "UPDATE `rates` SET `amount`=" + TextBox1.Text + " WHERE `ID`=" + ikk);
				ComboBox1.Text = "";
				ComboBox2.Text = "";
				TextBox1.Text = "";
				ComboBox1.Enabled = true;
				ComboBox2.Enabled = true;
				Button3.Text = "&Modify F3";
				Button2.Enabled = true;
				lev = level(ComboBox1.SelectedIndex);
				DataTable ggf = null;
				ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates WHERE rates.L_id =" + lev);
				DataGridView1.DataSource = ggf;
				int j = 0;
				int sum = 0;
				while (!(j == DataGridView1.Rows.Count)) {
					sum = sum + DataGridView1.Rows[i].Cells[3].Value;
					j = j + 1;
				}
				Label3.Text = "Total = Ksh " + sum;
				return;
			}
			if (DataGridView1.RowCount > 0) {
				i = DataGridView1.CurrentRow.Index;
				var _with1 = DataGridView1;
				h = _with1.Item(0, i).Value;
				ikk = h;
				ComboBox1.Text = _with1.Item(1, i).Value;
				ComboBox2.Text = _with1.Item(2, i).Value;
				TextBox1.Text = _with1.Item(3, i).Value;
				ComboBox1.Enabled = false;
				ComboBox2.Enabled = false;
				Button3.Text = "Save";
				Button2.Enabled = false;
			}

		}

		private void Button6_Click(object sender, EventArgs e)
		{
			 // ERROR: Not supported in C#: OnErrorStatement

			lev = level(ComboBox1.SelectedIndex);
			DataTable ggf = null;
			ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates WHERE rates.L_id =" + lev);
			DataGridView1.DataSource = ggf;
			int i = 0;
			int sum = 0;
			while (!(i == DataGridView1.Rows.Count)) {
				sum = sum + DataGridView1.Rows[i].Cells[3].Value;
				i = i + 1;
			}
			Label3.Text = "Total = Ksh " + sum;
		}

		private void TextBox1_KeyDown(object sender, KeyEventArgs e)
		{

		}

		private void TextBox1_KeyPress(object sender, KeyPressEventArgs e)
		{
			validate val = new validate();
			val.digits(sender, e);
		}


		private void TextBox1_TextChanged(object sender, EventArgs e)
		{
		}

		private void Button4_Click(object sender, EventArgs e)
		{
			int i = 0;
			string h = null;
			if (DataGridView1.RowCount > 0) {
				i = DataGridView1.CurrentRow.Index;
				var _with2 = DataGridView1;
				h = _with2.Item(0, i).Value;
				if (Interaction.MsgBox(" are you sure you want to delete" + Constants.vbCrLf + _with2.Item(1, i).Value + " " + _with2.Item(2, i).Value, Constants.vbYesNo, "confirm") == Constants.vbYes) {
					data.delete("DELETE FROM `rates` WHERE `ID`=" + h);
					lev = level(ComboBox1.SelectedIndex);
					DataTable ggf = null;
					ggf = data.executeSQL("SELECT ID, (SELECT description FROM class WHERE code=rates.L_id)'Class', (SELECT Type_Name FROM type WHERE  ID=rates.T_id)'Type', amount FROM rates WHERE rates.L_id =" + lev);
					DataGridView1.DataSource = ggf;
				}
			}
		}



		private void CheckBox1_CheckedChanged(object sender, EventArgs e)
		{
			if (CheckBox1.Checked == true) {
				ComboBox1.Enabled = false;
			} else {
				ComboBox1.Enabled = true;
			}
		}
		public fees_rates()
		{
			Load += fees_rates_Load;
			InitializeComponent();
		}
	}
}
